.github/workflows/build-and-push.yml

```diff
@@ -0,0 +1,93 @@
+name: Build, test and push a Docker image
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - v20[0-9][0-9].[01][0-9].[0-3][0-9]  # e.g. v2023.12.04
+      - v20[0-9][0-9].[01][0-9].[0-3][0-9]-[0-9]  # e.g. v2023.12.04-2
```

Comment:
Currently Antenna uses a `YYYY.MM.DD[-N]` tag scheme. We haven't changed that, yet. 

* [bug 1882879: changing Antenna tag scheme](https://bugzilla.mozilla.org/show_bug.cgi?id=1882879)
* [bug 1857938: changing Socorro tag scheme](https://bugzilla.mozilla.org/show_bug.cgi?id=1857938)

We should probably do those before setting up gha based deploys.